File: 1680-concatenation-of-consecutive-binary-numbers/1680-concatenation-of-consecutive-binary-numbers.cpp
class Solution {
public:
    int DToB(int num){
        int str=0;
        while(num){
            str++;
            num>>=1; 
        }   
        return str;
    }
    int concatenatedBinary(int n) {
        long long ans=0,mod=1000000007;
        for(int i=1;i<=n;i++){
            int l=DToB(i);
            ans=((ans<<l%mod)+i)%mod;
        }
        return ans;
    }
};

File: 1680-concatenation-of-consecutive-binary-numbers/1680-concatenation-of-consecutive-binary-numbers.py
class Solution:
    def concatenatedBinary(self, n: int) -> int:
        st=""
        for i  in range(1,n+1):
            s=str(bin(i))[2:]
            st+=s
        val=int(st,2)
        mod=(10**9)+7
        return val%mod